lib/logutils/logutils_test.go

```diff
@@ -168,15 +175,45 @@ var _ = Describe("BackgroundHook log flushing tests", func() {
 			Level:   log.DebugLevel,
 			Channel: c,
 		}
-		bh = NewBackgroundHook(log.AllLevels, log.DebugLevel, []*Destination{testDest}, counter)
+		bh = NewBackgroundHook(log.AllLevels, log.DebugLevel, []*Destination{testDest}, counter, hookOpts...)
 
 		logger = log.New()
+		logger.AddHook(ContextHook{})
 		logger.AddHook(bh)
+		logger.SetLevel(log.DebugLevel)
 
 		// Suppress the output of this logger.
 		logger.Out = &NullWriter{}
 	})
 
+	It("should let debug logs through by default", func() {
+		logger.Debug("Hello")
+		var ql QueuedLog
+		Eventually(c).Should(Receive(&ql))
+		Expect(string(ql.Message)).To(ContainSubstring("level=debug msg=Hello"))
+	})
+
+	Describe("with a regex set", func() {
+		BeforeEach(func() {
+			hookOpts = append(hookOpts, WithDebugFileRegexp(regexp.MustCompile("another_file_for_test")))
+		})
+
+		It("should filter debug logs", func() {
+			logger.Debug("Hello")
+			debugFromAnotherFile(logger, "What?")
+			var ql QueuedLog
+			Eventually(c).Should(Receive(&ql))
+			Expect(string(ql.Message)).To(ContainSubstring(`level=debug msg="What?"`))
```

Comment:
Should there be a check here for not getting a "Hello" log?

Reply:
I was relying on implicitly receiving the next log to signal that the first one was filtered.  But, no harm in being explicit so I added an extra line.